src/material/dialog/dialog-container.ts

```diff
@@ -117,6 +126,59 @@ export class MatDialogContainer extends CdkDialogContainer<MatDialogConfig> impl
     );
   }
 
+  /** Get userAgent to check for useragent operating system */
+  private _getUserPlatform = (): string => {
```

Comment:
We have a cdk/platform package that we use for this type of thing

Reply:
The documentation I found for the Platform CDK seems to only check for iOS as opposed to macOS. Updated the Platform CDK to check for mac, windows, and linux and replaced window.navigator.userAgent with using Platform CDK as recommended instead.

Reply:
See this commit for reference of Platform CDK Update: https://github.com/angular/components/pull/29264/commits/50458f84960c442025fdd2a4ec886f8d6d0fbae3